src/yachtzee/buffs.ts

```diff
@@ -0,0 +1,215 @@
+        "blue"
+      );
+      if (!simOnly) {
+        cliExecute("pillkeeper extend");
+        acquire(1, bestPotion.potion, profit + price);
+        bestPotion.use(1);
+      } else excludedEffects.add(bestPotion.effect());
+    }
+  }
+
+  for (const effect of getActiveEffects()) {
+    for (const excluded of mutuallyExclusive.get(effect) ?? []) {
+      excludedEffects.add(excluded);
+    }
+  }
+
+  const testPotions = farmingPotions
+    .filter(
+      (potion) =>
+        haveEffect(potion.effect()) + PYECOffset * toInt(haveEffect(potion.effect()) > 0) <
+          yachtzeeTurns && yachtzeePotionProfits(potion, yachtzeeTurns) > 0
+    )
+    .sort(
+      (left, right) =>
+        yachtzeePotionProfits(right, yachtzeeTurns) - yachtzeePotionProfits(left, yachtzeeTurns)
+    );
+
+  for (const potion of testPotions) {
+    const effect = potion.effect();
+    const price = potion.price(true);
+    if (
+      haveEffect(effect) + PYECOffset * toInt(haveEffect(effect) > 0) >= yachtzeeTurns ||
+      price > myMeat()
+    ) {
+      continue;
+    }
+    if (!excludedEffects.has(effect)) {
+      let tries = 0;
+      while (haveEffect(effect) + PYECOffset * toInt(haveEffect(effect) > 0) < yachtzeeTurns) {
+        tries++;
+        print(`Considering effect ${effect} from source ${potion.potion}`, "blue");
+        const profit = yachtzeePotionProfits(potion, yachtzeeTurns);
+        if (profit < 0) break;
+        const nPotions = have(effect)
+          ? clamp(
+              Math.floor(
+                (yachtzeeTurns - haveEffect(effect) - PYECOffset) / potion.effectDuration()
+              ),
+              1,
+              Math.max(1, yachtzeeTurns - PYECOffset)
+            )
+          : 1;
+
+        totalProfits += nPotions * profit;
+        print(
+          `Expected to profit ${nPotions * profit} meat from using ${nPotions} ${
+            potion.potion
+          } @ price ${price} meat each`,
+          "blue"
+        );
+        if (!simOnly) {
+          acquire(nPotions, potion.potion, profit + price, false);
+          if (itemAmount(potion.potion) < 1) break;
+          if (isSong(effect) && !have(effect)) {
+            for (const song of getActiveSongs()) {
+              const slot = Mood.defaultOptions.songSlots.find((slot) => slot.includes(song));
+              if (!slot || slot.includes(effect)) {
+                cliExecute(`shrug ${song}`);
+              }
+            }
+          }
+          if (
+            !potion.use(Math.min(itemAmount(potion.potion), nPotions)) ||
+            tries >= 5 * Math.ceil(yachtzeeTurns / potion.effectDuration())
+          ) {
+            break;
+          }
+        } else break;
+      }
+      if (have(effect) || simOnly) {
+        for (const excluded of mutuallyExclusive.get(effect) ?? []) {
+          excludedEffects.add(excluded);
+        }
+      }
+    }
+  }
+
+  if (!simOnly) {
+    variableMeatPotionsSetup(yachtzeeTurns, expectedEmbezzlers);
```

Comment:
I have yet to add the profits from variable meat potions to the `totalProfits`, similar to Greenspans, but I think this is a much easier task than the latter.
Not a major issue, but you can consider adding it if you'd like (note that we would then need to pass a simOnly flag into `variableMeatPotionsSetup` because as of now it automatically triggers the acquisition and usage of the potions)

Reply:
Out of scope of this PR